crates/router/src/core/unified_connector_service.rs

```diff
@@ -0,0 +1,265 @@
+        |value: &str,
+         context: &str|
+         -> CustomResult<MetadataValue<_>, UnifiedConnectorServiceError> {
+            value
+                .parse::<MetadataValue<_>>()
+                .change_context(UnifiedConnectorServiceError::HeaderInjectionFailed(
+                    context.to_string(),
+                ))
+                .attach_printable(format!(
+                    "Failed to parse metadata value for {context}: {value}"
+                ))
+        };
+
+    match &auth_type {
+        ConnectorAuthType::SignatureKey {
+            api_key,
+            key1,
+            api_secret,
+        } => {
+            metadata.append(
+                consts::UCS_HEADER_AUTH_TYPE,
+                parse_metadata_value(consts::UCS_AUTH_SIGNATURE_KEY, consts::UCS_HEADER_AUTH_TYPE)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_KEY,
+                parse_metadata_value(api_key.peek(), consts::UCS_HEADER_API_KEY)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_KEY1,
+                parse_metadata_value(key1.peek(), consts::UCS_HEADER_KEY1)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_SECRET,
+                parse_metadata_value(api_secret.peek(), consts::UCS_HEADER_API_SECRET)?,
+            );
+        }
+        ConnectorAuthType::BodyKey { api_key, key1 } => {
+            metadata.append(
+                consts::UCS_HEADER_AUTH_TYPE,
+                parse_metadata_value(consts::UCS_AUTH_BODY_KEY, consts::UCS_HEADER_AUTH_TYPE)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_KEY,
+                parse_metadata_value(api_key.peek(), consts::UCS_HEADER_API_KEY)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_KEY1,
+                parse_metadata_value(key1.peek(), consts::UCS_HEADER_KEY1)?,
+            );
+        }
+        ConnectorAuthType::HeaderKey { api_key } => {
+            metadata.append(
+                consts::UCS_HEADER_AUTH_TYPE,
+                parse_metadata_value(consts::UCS_AUTH_HEADER_KEY, consts::UCS_HEADER_AUTH_TYPE)?,
+            );
+            metadata.append(
+                consts::UCS_HEADER_API_KEY,
+                parse_metadata_value(api_key.peek(), consts::UCS_HEADER_API_KEY)?,
+            );
+        }
+        _ => {
+            return Err(UnifiedConnectorServiceError::FailedToObtainAuthType)
+                .attach_printable("Unsupported ConnectorAuthType for header injection")?;
+        }
+    }
+
+    Ok(metadata)
+}
+
+pub fn handle_unified_connector_service_response_for_payment_authorize(
+    response: PaymentsAuthorizeResponse,
+) -> CustomResult<
+    (AttemptStatus, Result<PaymentsResponseData, ErrorResponse>),
+    UnifiedConnectorServiceError,
+> {
+    let status = AttemptStatus::foreign_try_from(response.status())?;
+
+    let router_data_response = match status {
```

Comment:
Wherever it is None, is these fields not available in PaymentsAuthorizeResponse?